lib/icinga/apiactions.cpp

```diff
@@ -200,7 +200,7 @@ Dictionary::Ptr ApiActions::AcknowledgeProblem(const ConfigObject::Ptr& object,
 	if (params->Contains("sticky"))
 		sticky = AcknowledgementSticky;
```

Comment:
That should check for a boolean, not just for the attribute being set. 

Similar fix to be applied: https://github.com/Icinga/icinga2/commit/81965238d0cfd4e9ab8ef2cb41f8cb27ae2de234

```
if (params->Contains("sticky") && HttpUtility::GetLastParameter(params, "sticky"))
```